kahuna/public/js/components/gu-lazy-gallery/gu-lazy-gallery.js

```diff
@@ -0,0 +1,144 @@
+import angular from 'angular';
+import Rx from 'rx';
+import 'rx-dom';
+
+import template from './gu-lazy-gallery.html!text';
+
+import '../../util/rx';
+
+export var lazyGallery = angular.module('gu.lazyGallery', ['util.rx']);
+
+function asInt(string) {
+    return parseInt(string, 10);
+}
+
+lazyGallery.controller('GuLazyGalleryCtrl', ['$scope', 'subscribe$', function($scope, subscribe$) {
+    let ctrl = this;
+    ctrl.canGoNext = false;
+    ctrl.canGoPrev = false;
+
+    ctrl.init = function({items$, preloadedItems$, currentIndex$}) {
+        const itemsCount$ = items$.map(items => items.length).distinctUntilChanged();
+
+        const buttonCommands$ = new Rx.BehaviorSubject('galleryStart');
```

Comment:
Hmm true, I'll take a look. When I initialise in the creation block I can add something straight to the stream.